### PR TITLE
Update join.py

### PR DESCRIPTION
**Pull request by ccl-core**
To solve signature-mismatch error

### CODE REVIEW

File: python/mlcroissant/mlcroissant/_src/operation_graph/operations/join.py [hunk 1]
```diff
@@ -15,7 +17,7 @@ class Join(Operation):
 
     node: RecordSet
 
-    def __call__(self, *args: pd.Series) -> pd.Series:
+    def __call__(self, *args: pd.Series) -> Union[pd.Series, None]:
```
Comment (marcenacp) [inline python/mlcroissant/mlcroissant/_src/operation_graph/operations/join.py:20]: nit: `pd.Series | None`

File: python/mlcroissant/mlcroissant/_src/operation_graph/operations/filter.py [hunk 2]
```diff
@@ -23,13 +23,13 @@ def __call__(self, *dirs: Path) -> list[Path]:
         """See class' docstring."""
         includes = fnmatch.translate(self.node.includes)
         includes_re = re.compile(includes)
-        included_files: list[epath.Path] = []
+        included_files: list[Path] = []
```
Comment (marcenacp) [inline python/mlcroissant/mlcroissant/_src/operation_graph/operations/filter.py:26]: nit: `included_files` could now be renamed to `included_paths`.

(`Path` used to be named `File`. I think there's this discrepancy in several places.)

**Review by marcenacp — APPROVED** [review]
Great, thank you!

**Comment by marcenacp** [comment]
In order to solve the `Python format` test, you can try to use VS Code to format your code.

**Comment by ccl-core** [comment]
Closing because it was already addressed in the PRs submitted between October 19-20

**Pull request closed without merging**